src/sagemaker_containers/environment.py

```diff
@@ -250,6 +226,155 @@ def __iter__(self):
+
+            num_gpu (int): The number of gpus available in the current container.
+
+            num_cpu (int): The number of cpus available in the current container.
+        """
+        self._current_host = current_host
+        self._num_gpu = num_gpu
+        self._num_cpu = num_cpu
+        self._module_name = self._parse_module_name(module_name)
+        self._module_dir = module_dir
+        self._enable_metrics = enable_metrics
+        self._log_level = log_level
+
+    @property
+    def current_host(self):  # type: () -> str
+        """The name of the current container on the container network. For example, 'algo-1'.
+        Returns:
+            (str): current host.
+        """
+        return self._current_host
+
+    @property
+    def num_gpu(self):  # type: () -> int
+        """The number of gpus available in the current container.
+
+        Returns:
+            (int): number of gpus available in the current container.
+        """
+        return self._num_gpu
+
+    @property
+    def num_cpu(self):  # type: () -> int
+        """The number of cpus available in the current container.
+
+        Returns:
+            (int): number of cpus available in the current container.
+        """
+        return self._num_cpu
+
+    @property
+    def module_name(self):  # type: () -> str
+        """The name of the user provided module.
+
+        Returns:
+            (str): name of the user provided module
+        """
+        return self._module_name
+
+    @property
+    def module_dir(self):  # type: () -> str
+        """The full path location of the user provided module.
+
+        Returns:
+            (str): full path location of the user provided module.
+        """
+        return self._module_dir
+
+    @property
+    def enable_metrics(self):  # type: () -> bool
+        """Whether metrics should be executed in the environment or not.
+
+        Returns:
+            (bool): representing whether metrics should be executed or not.
+        """
+        return self._enable_metrics
+
+    @property
+    def log_level(self):  # type: () -> int
+        """Environment logging level.
+
+        Returns:
+            (int): environment logging level.
+        """
+        return self._log_level
+
+    @classmethod
+    @abstractmethod
+    def create(cls, session=None):  # type: (boto3.Session) -> Environment
+        """
+        Returns: an instance of `Environment`
+        """
+        pass
+
+    @staticmethod
+    def _parse_module_name(program_param):
+        """Given a module name or a script name, Returns the module name.
+
+        This function is used for backwards compatibility.
+
+        Args:
+            program_param (str): Module or script name.
+
+        Returns:
+            (str): Module name
+        """
+        if program_param.endswith('.py'):
+            return program_param[:-3]
+        return program_param
+
+
+class TrainingEnvironment(Environment):
```

Comment:
to me, "Trainer" would imply that the environment is actually performing the training, whereas "Training" implies that the environment is the place where training occurs.

Reply:
That sounds ... convincing. Glad to have someone in the team that look so deep in the meaning of the names. I just liked it more because it is shorter, which is recommended by PEP 8 https://www.python.org/dev/peps/pep-0008/#package-and-module-names

These classes are not modules... yet. But I my understanding is that it will happen soon. All that being said, I am happy to keep the current naming.

Reply:
it's a difference of one letter in this case :joy:

I think there are more opportunities to cut letters by changing "Environment" to "Env" if the name is too long.

Reply:
My brain is playing tricks on me, somehow I thought I was saving 3 letters.

I thought about TraininingEnv before as well, but I am not sold on that.

Another option:

```
smc.trainining.Environment() and smc.serving.Environment()
```